test/jdk/jdk/nio/zipfs/TestLocOffsetFromZip64EF.java

```diff
@@ -143,88 +127,70 @@ public void walkZipFileTest() throws IOException {
     }
 
     /**
-     * Create the files that will be added to the ZIP file
-     * @throws IOException if there is a problem  creating the files
-     */
-    private static void createFiles() throws IOException {
-        try (RandomAccessFile file = new RandomAccessFile(LARGE_FILE_NAME, "rw")
-        ) {
-            System.out.printf("Creating %s%n", LARGE_FILE_NAME);
-            file.setLength(LARGE_FILE_SIZE);
-            System.out.printf("Creating %s%n", SMALL_FILE_NAME);
-            Files.writeString(Path.of(SMALL_FILE_NAME), "Hello");
-        }
-    }
-
-    /**
-     * Utility method to execute a ProcessBuilder command
-     * @param pb ProcessBuilder to execute
-     * @return The Result(s) from the ProcessBuilder execution
+     * Create a ZIP file where the second entry is large enough to output a
+     * Zip64 CEN entry where the LOC offset is found inside a Zip64 Extra field.
      */
-    private static Result run(ProcessBuilder pb) {
-        Process p;
-        System.out.printf("Running: %s%n", pb.command());
-        try {
-            p = pb.start();
-        } catch (IOException e) {
-            throw new RuntimeException(
-                    format("Couldn't start process '%s'", pb.command()), e);
-        }
-
-        String output;
-        try {
-            output = toString(p.getInputStream(), p.getErrorStream());
-        } catch (IOException e) {
-            throw new RuntimeException(
-                    format("Couldn't read process output '%s'", pb.command()), e);
-        }
-
-        try {
-            p.waitFor();
-        } catch (InterruptedException e) {
-            throw new RuntimeException(
-                    format("Process hasn't finished '%s'", pb.command()), e);
+    public void createZipWithZip64Ext() throws IOException {
+        // Make a ZIP with two entries
+        try (FileOutputStream fileOutputStream = new FileOutputStream(new File(ZIP_FILE_NAME));
+             ZipOutputStream zo = new ZipOutputStream(new SparseOutputStream(fileOutputStream))) {
```

Comment:
Can you use FileChannel.open and specify SPARSE in the set of open options, I think that would make it clearer that the file is sparse and remove the need for SparseOutputStream.

Reply:
Not sure I understand. I could use FileChannel.open with SPARSE (which btw is ignored by the implementation:), but that doesn't give me an OutputStream I can pass to ZipOutputStream?

I could use Channels.newOutputStream, but that would not create holes in the sparse files?

My current thinking is that we need the SparseOutputStream to detect that ZIpOutputStream is writing empty bytes and replace that with `channel.position(channel.position() + len)`

Reply:
> Not sure I understand. I could use FileChannel.open with SPARSE (which btw is ignored by the implementation:)

That is true on platforms where files can be sparse by default. On Windows you'll see this translates to create the file as sparse, it needs to be opt-in when creating the file.

Reply:
I opted to create a small-sized Zip64 file instead (see below). This file does not need to be sparse.

Reply:
> That is true on platforms where files can be sparse by default. On Windows you'll see this translates to create the file as sparse, it needs to be opt-in when creating the file.

Thanks for this info, Alan!

While this PR no longer makes use of sparse files, I have applied your comments about FileChannel.open for NTFS in another PR (which does make use of sparse files):

https://github.com/openjdk/jdk/pull/12948